### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by Dasari-Jay-Prakash12**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,94 @@
+License for everything not in third_party and not otherwise marked:
```
Comment (Himanshu-0123) [inline LICENSE]: update the license file

File: entry/hvigorfile.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Script for compiling build behavior. It is built in the build plug-in and cannot be modified currently.
```
Comment (Himanshu-0123) [inline entry/hvigorfile.js:15]: add copyright text

File: entry/src/main/ets/Application/AbilityStage.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import AbilityStage from "@ohos.application.AbilityStage"
```
Comment (Himanshu-0123) [inline entry/src/main/ets/Application/AbilityStage.ts:15]: add copyright text

File: entry/src/main/ets/MainAbility/MainAbility.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import Ability from '@ohos.application.Ability'
```
Comment (Himanshu-0123) [inline entry/src/main/ets/MainAbility/MainAbility.ts:15]: add copyright text

File: entry/src/ohosTest/ets/TestAbility/TestAbility.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import Ability from '@ohos.application.Ability'
```
Comment (Himanshu-0123) [inline entry/src/ohosTest/ets/TestAbility/TestAbility.ts:15]: add copyright text

File: entry/src/ohosTest/ets/TestAbility/pages/index.ets [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import router from '@ohos.router';
```
Comment (Himanshu-0123) [inline entry/src/ohosTest/ets/TestAbility/pages/index.ets:15]: add copyright text

File: entry/src/ohosTest/ets/TestRunner/OpenHarmonyTestRunner.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
+import TestRunner from '@ohos.application.testRunner'
```
Comment (Himanshu-0123) [inline entry/src/ohosTest/ets/TestRunner/OpenHarmonyTestRunner.ts:15]: add copyright text

File: entry/src/ohosTest/ets/test/Ability.test.ets [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { describe, beforeAll, beforeEach, afterEach, afterAll, it, expect } from 'hypium/index'
```
Comment (Himanshu-0123) [inline entry/src/ohosTest/ets/test/Ability.test.ets:15]: add copyright text

File: entry/src/ohosTest/ets/test/List.test.ets [hunk 9]
```diff
@@ -0,0 +1,5 @@
+import abilityTest from './Ability.test'
```
Comment (Himanshu-0123) [inline entry/src/ohosTest/ets/test/List.test.ets:15]: add copyright text

File: hvigorfile.js [hunk 10]
```diff
@@ -0,0 +1,2 @@
+// Script for compiling build behavior. It is built in the build plug-in and cannot be modified currently.
```
Comment (Himanshu-0123) [inline hvigorfile.js:15]: add copyright text

File: simpletransition/hvigorfile.js [hunk 11]
```diff
@@ -0,0 +1,3 @@
+// Script for compiling build behavior. It is built in the build plug-in and cannot be modified currently.
```
Comment (Himanshu-0123) [inline simpletransition/hvigorfile.js:15]: add copyright text

File: simpletransition/index.ets [hunk 12]
```diff
@@ -0,0 +1,2 @@
+export { SimpleTransition } from './src/main/ets/components/MainPage/Spring-Transition'
```
Comment (Himanshu-0123) [inline simpletransition/index.ets:15]: add copyright text

File: simpletransition/src/main/ets/components/MainPage/Spring-Transition.ets [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+import Curves from '@ohos.curves'
+import { Transition } from './CommonProperties'
+
+@Component
+export struct SimpleTransition {
+  @State toggle: boolean = true
+  model: Transition.Model= undefined
+
```
Comment (Himanshu-0123) [inline simpletransition/src/main/ets/components/MainPage/Spring-Transition.ets:23]: add default values for toggle & model variable in aboutToAppear() if null & undefined are passed

Comment (Dasari-Jay-Prakash12) [reply simpletransition/src/main/ets/components/MainPage/Spring-Transition.ets:23]: done

File: simpletransition/index.ets (hunk 12 above)
Comment (Dasari-Jay-Prakash12) [reply simpletransition/index.ets:15]: done

File: simpletransition/hvigorfile.js (hunk 11 above)
Comment (Dasari-Jay-Prakash12) [reply simpletransition/hvigorfile.js:15]: done

File: hvigorfile.js (hunk 10 above)
Comment (Dasari-Jay-Prakash12) [reply hvigorfile.js:15]: done

File: entry/src/ohosTest/ets/test/List.test.ets (hunk 9 above)
Comment (Dasari-Jay-Prakash12) [reply entry/src/ohosTest/ets/test/List.test.ets:15]: done

File: entry/src/ohosTest/ets/test/Ability.test.ets (hunk 8 above)
Comment (Dasari-Jay-Prakash12) [reply entry/src/ohosTest/ets/test/Ability.test.ets:15]: done

File: entry/src/ohosTest/ets/TestRunner/OpenHarmonyTestRunner.ts (hunk 7 above)
Comment (Dasari-Jay-Prakash12) [reply entry/src/ohosTest/ets/TestRunner/OpenHarmonyTestRunner.ts:15]: done

File: entry/src/ohosTest/ets/TestAbility/pages/index.ets (hunk 6 above)
Comment (Dasari-Jay-Prakash12) [reply entry/src/ohosTest/ets/TestAbility/pages/index.ets:15]: done

File: entry/src/ohosTest/ets/TestAbility/TestAbility.ts (hunk 5 above)
Comment (Dasari-Jay-Prakash12) [reply entry/src/ohosTest/ets/TestAbility/TestAbility.ts:15]: done

File: entry/src/main/ets/MainAbility/MainAbility.ts (hunk 4 above)
Comment (Dasari-Jay-Prakash12) [reply entry/src/main/ets/MainAbility/MainAbility.ts:15]: done

File: entry/src/main/ets/Application/AbilityStage.ts (hunk 3 above)
Comment (Dasari-Jay-Prakash12) [reply entry/src/main/ets/Application/AbilityStage.ts:15]: done

File: entry/hvigorfile.js (hunk 2 above)
Comment (Dasari-Jay-Prakash12) [reply entry/hvigorfile.js:15]: done

File: LICENSE (hunk 1 above)
Comment (Dasari-Jay-Prakash12) [reply LICENSE]: done

File: simpletransition/src/main/ets/components/MainPage/Spring-Transition.ets [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+  aboutToAppear() {
+    if(this.model == undefined || this.model==null){
+      this.model =  new Transition.Model()
+    }
+    if(this.toggle == undefined || this.toggle ==null){
+      this.toggle = true
+    }
+  }
+
+  aboutToAppear() {
```
Comment (Himanshu-0123) [inline simpletransition/src/main/ets/components/MainPage/Spring-Transition.ets:24]: aboutToAppear is added twice remove one

Comment (Dasari-Jay-Prakash12) [reply simpletransition/src/main/ets/components/MainPage/Spring-Transition.ets:24]: Done

**Pull request merged**